Diabetes/projects/diabetes_project_1.html

```diff
@@ -0,0 +1,87 @@
+<html>
+
+<head>
+  <script src='https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js'></script>
+  <script type='text/javascript'>
+    $(document).ready(function () {
+      $('body').append(`<p id="prediction">Prediction: <p>`);
+
+      $.fn.serializeObject = function () {
+        var object = {};
+        $.each(this.serializeArray(), function () {
+          if (object[this.name] !== undefined) {
+            [object[this.name]].push(this.value || '');
+          } else {
+            object[this.name] = this.value || '';
+          }
+        });
+        return object;
+      };
+
+      var DATA;
+      function predict(data = DATA) {
+        console.log('data', DATA);
+        var ENDPOINT = $('[name=url]').val();
+        return fetch(ENDPOINT, {
+          method: 'POST',
+          body: data,
+        })
+          .then((res) => res.json())
+          .then((resJSON) => {
+            console.log('resJSON', resJSON);
+            resJSON = JSON.parse(resJSON['body']);
+            var innerHTML = $('#prediction').filter('p').text();
+            var label = innerHTML.substring(innerHTML.indexOf('Prediction:') + 'Prediction:'.length);
+            $('#prediction').html($('#prediction').html().replace(label, resJSON['predicted_label']));
+          })
+          .catch(err => console.log('err', err));
+      };
+
+      $('#PredictButton').click(function (data = DATA) {
+        DATA = JSON.stringify($('form').serializeObject());
+        predict();
+      });
+    });
+  </script>
+</head>
+
+<body>
+
+  <p>Diabetes Prediction Service</p>
+
+  Url:<br>
+  <input type="text" name="url">
+  <br>
+  <form id="myform">
+    Pregnancies:<br>
```

Comment:
use jquery to map over an array with the specifications to create this form input. This way, if you need to make more forms, you just copy the function instead of rewriting this html code.

example:
```
var FIELDS = [
  {
    label: 'Pregancies:',
    type: 'text',
    name: 'Pregancies',
    value: '6',
  },
  {
    label: 'Glucose:',
    type: 'number',
    name: 'Glucose',
    value: '148',
  },
  etc...
];
```

then have function that can generate the html for each field.

```
makeField(field) {
  return `
    ${field.label} <br/ >
    <input type=${field.type} name=${field.name} value=${field.value} />
    <br />
 `;
}
```

Then you can do this to append the html elements to the form
```
FIELDS.forEach(field => $('#myform').append(makeField(field)));
```

Reply:
Ahh thanks for the feedback!